### DISCUSSION TITLE
Vehicle window - show the most important information on top

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#12197] [Features]
Recently you introduced a change to the order of information in the bottom bar, so that the player always sees the vehicle's current speed. For me, this is a very good change. It was a bit tiring to have to stretch the vehicle's window every time I wanted to see its current speed. :) But... This arrangement won't always be the best either. For example, if it is a city bus, in the case of cities with longer names you will never see which station it is going to. The problem here is not only the speed information, which has gained priority, but also the "Heading for..." information that takes up valuable space.

The solution I would like to present is not a new idea, it is already several years old, but I still think it could be a nice solution to the above and several other problems.
The general idea is to bring to light the four most important information about the vehicle that may be important to the player:
- current speed
- loading percentage
- number of failures if enabled
- timetable delay if set

This information would be placed in the corners of the window as overlaid text (preferably with a shaded background for better readability).
Currently, you no longer have to stretch the window to see the speed, but to see other information you still have to open other windows and the appropriate tabs in them. When you want to check a larger number of vehicles, it becomes quite tedious. Additionally, by moving the speed information and removing the unnecessary "Heading for" from the main bar, the player would more often see the full names of the directions the vehicles are taking.
The presentation was supposed to cover the vehicles window in general, so there are also two unrelated ideas about a different style of icons and adding a button to show the vehicle's route on the game map or a shortcut to the list of sharing vehicles - here the icon itself would also inform whether the vehicle shares orders.

![Main vehicle window - description v1 4](https://github.com/OpenTTD/OpenTTD/assets/67662046/094ecfe3-9139-4d7b-9771-1c6dd88d0aff)

Comparison with current and previous layout
![Main vehicle window v1 4](https://github.com/OpenTTD/OpenTTD/assets/67662046/b0f71716-815e-4057-8814-e7a592af5867)

**Comment by zephyris** [comment]
It looks nice enough. My main comment is to keep things synchronised. If key information is to be overlaid on the viewport of vehicles, then the same should probably done for eg. town windows, industry windows. Users will expect to see key information in the same place in similar window layouts.

**Reply by LC-Zorg** [reply]
I think there would be a problem with synchronization here. ;) In the case of city and industry windows, such a change would be simply unnecessary. In my opinion, their current format doesn't require such changes. There, unlike vehicle windows, all information is visible. Maybe the only thing missing is more accessible information about the company's rating. Window size reduction? For some enterprise sets and scripts, given the amount of information, this form would not be good. In the case of vehicles, there is a maximum of 4 of these important pieces of information. I had previously considered a different layout, but the one with data placed in the corners, which did not increase the size of the window or cover the vehicle, seemed the most functional to me.

One of the early, alternative versions
![Main vehicle window v0 1](https://github.com/OpenTTD/OpenTTD/assets/67662046/87e22bfe-f065-466f-8987-b3c8e4824c40)

**Comment by nielsmh** [comment]
There is a minor technical challenge due to how viewports are drawn. Right now there isn't a good/reliable way to draw text on top of the viewport for the vehicle, with transparent background.
All text currently drawn on top of world viewports is bound to world space, rather than window space. This also means that all that text (signs, town names, income figures) appears in every viewport showing that area, rather than only that one single viewport.

So the text and viewport rendering would need to be update to be able to support this idea.
